src/controller/python/chip/clusters/Attribute.py

```diff
@@ -719,22 +720,33 @@ def _handleSubscriptionEstablished(self, subscriptionId):
             logging.info("Re-subscription succeeded!")
             if self._subscription_handler._onResubscriptionSucceededCb is not None:
                 if (self._subscription_handler._onResubscriptionSucceededCb_isAsync):
-                    self._event_loop.create_task(
-                        self._subscription_handler._onResubscriptionSucceededCb(self._subscription_handler))
+                    if self._event_loop.is_closed():
+                        asyncio.run(
+                            self._subscription_handler._onResubscriptionSucceededCb(self._subscription_handler))
+                    else:
+                        self._event_loop.create_task(
```

Comment:
it seems not to be right, if event_loop is open, then create a new loop?  should we switch the else's logic to if part?

Reply:
Pretty sure it is correct?  asyncio.run creates a new new loop.  create_task adds to an existing loop.

https://docs.python.org/3/library/asyncio-task.html#asyncio.run

> This function always creates a new event loop and closes it at the end.

https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.loop.create_task

> Schedule the execution of [coroutine](https://docs.python.org/3/library/asyncio-task.html#coroutine) coro.

```python
                # async function so will return typing.Coroutine
                if (self._subscription_handler._onResubscriptionSucceededCb_isAsync):
                    if self._event_loop.is_closed():  # loop passed to object initialization is closed
                        asyncio.run(  # so create a new loop, run this method, block until it returns
                            self._subscription_handler._onResubscriptionSucceededCb(self._subscription_handler))
                    else:  # loop passed to object initialization is open
                        self._event_loop.create_task(  # add typing.Coroutine onto existing loop
                            self._subscription_handler._onResubscriptionSucceededCb(self._subscription_handler))
```

Reply:
I see, thanks for the clarification LGTM.